src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -0,0 +1,153 @@
+    const mask: Set<number> = new Set();
+    // Represent [R,G,B,A] attributes of initial pixel
+    const originalPixel: Array<number> =
+        this.dataArrayToRgba(imgData, xCoord, yCoord);
+
+    visit.push([xCoord, yCoord]);
+    // Convert [x,y] format coord to 1-D equivalent of imgData.data (DataArray)
+    let indexAsDataArray: number =
+        this.coordToDataArrayIndex(xCoord, yCoord, imgData.width);
+    visited.add(indexAsDataArray);
+
+    // Loop until no more adjacent pixels within tolerance level
+    while(visit.length != 0) {
+      let coord: Array<number> = visit.pop();
+      // Unpack coord
+      let x: number = coord[0];
+      let y: number = coord[1];
+
+      // Operational part of while-loop
+      mask.add(this.coordToDataArrayIndex(x, y, imgData.width));
+
+      // Loop part of while-loop
+
+      // Get coords of adjacent pixels
+      let neighbors: Array<Array<number>> =
+          [[x - 1, y], [x + 1, y], [x, y - 1], [x, y + 1]];
+      // Add coords of adjacent pixels to the heap
+      for (let neighborPixel of neighbors) {
+        if (this.getIsMask(originalPixel, imgData, neighborPixel, tolerance,
+            visited)) {
+          visit.push(neighborPixel);
+        }
+      }
+    }  // end of while loop
+
+    return mask;
+  }
+
+  /**First check that @pixelCoord is within range of the image bounds.
+   * Then, judge current pixel's RGBA against original pixel's RGBA to
+   * see if it can still be part of the mask (using tolerance criteria).
+   */
+  getIsMask(originalPixel: Array<number>, imgData: ImageData,
+      pixelCoord: Array<number>, tolerance: number,
+      visited: Set<number>): boolean {
+    let curX: number = pixelCoord[0];
+    let curY: number = pixelCoord[1];
+    let imgWidth: number = imgData.width;
+
+    // Preface; check if pixel is valid (indexing errs and repeat values)
+    let isValid: boolean =
+        this.getIsValid(imgWidth, imgData.height, curX, curY, visited);
+    if (!isValid) {
+      // Automatically not in mask b/c failed vailidity test
+      return false;
+    }
+
+    visited.add(this.coordToDataArrayIndex(curX, curY, imgWidth));
+
+    // Get array of attributes of current pixel
+    let curPixel: Array<number> = this.dataArrayToRgba(imgData, curX, curY);
+
+    // All attributes of the pixel (R,G,B, and A) must be within tolerance level
+    for (let i = 0; i < 4; i++) {
+      let upperTolerance: boolean = curPixel[i] > originalPixel[i] + tolerance;
+      let lowerTolerance: boolean = curPixel[i] < originalPixel[i] - tolerance;
+      if (upperTolerance || lowerTolerance) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  // Check if @pixelCoord is in bounds and makes sure it's not a repeat coord
+  getIsValid(imgWidth: number, imgHeight: number, curX: number, curY: number,
+      visited: Set<number>): boolean {
+    if (this.isInBounds(imgWidth, imgHeight, curX, curY) &&
+        this.notVisited(imgWidth, curX, curY, visited)) {
```

Comment:
`return this.isInBounds(imgWidth, imgHeight, curX, curY) && this.notVisited(imgWidth, curX, curY, visited);`